### PR TITLE
Add Jaccard Loss

### PR DESCRIPTION
**Pull request by innat**
Close https://github.com/keras-team/keras-cv/issues/341

### CODE REVIEW

File: keras_cv/losses/jaccard.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        self.label_smoothing = label_smoothing
+        self.class_ids = class_ids
+        self.axis = axis
+        self.epsilon = epsilon
+
+    def _smooth_labels(self, y_true, y_pred, label_smoothing):
+        num_classes = tf.cast(tf.shape(y_true)[-1], y_pred.dtype)
+        return y_true * (1.0 - label_smoothing) + (label_smoothing / num_classes)
+
+    def call(self, y_true, y_pred):
```
Comment (tanzhenyu) [inline keras_cv/losses/jaccard.py:59]: Please add unit tests for numerical behavior

**Review by tanzhenyu — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/losses/jaccard.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            matrix = tf.gather(matrix, indices, axis=-1)
+        else:
+            matrix = tf.gather(matrix, indices, axis=1)
+        gathered_channels.append(matrix)
+
+    return gathered_channels
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class Jaccard(keras.losses.Loss):
```
[the remark this answers is not in the document]
Comment (innat) [reply keras_cv/losses/jaccard.py:37]: @tanzhenyu thanks for the feedback. I will update this PR. I'm waiting to make this PR https://github.com/keras-team/keras-cv/pull/371 ready first.

**Comment by LukeWood** [comment]
@innat will you be finishing this?

If so, you are welcome to re-open, but at the moment I think this PR is quite stale.
